modules/persistent-storage-csi-efs-sts.adoc

```diff
@@ -0,0 +1,94 @@
+// Module included in the following assemblies:
+//
+// * storage/persistent_storage/persistent-storage-csi-aws-efs.adoc
+
+:_content-type: PROCEDURE
+[id="efs-sts_{context}"]
+= Configuring AWS EFS CSI Driver Operator with Secure Token Service
+
+This procedure explains how to configure the AWS EFS CSI Driver Operator with {product-title} on AWS Secure Token Service (STS).
+
+You should perform this procedure after the AWS EFS CSI Operator is installed (xref:../../storage/container_storage_interface/persistent-storage-csi-aws-efs.adoc#persistent-storage-csi-olm-operator-install_persistent-storage-csi-aws-efs[Installing the AWS EFS CSI Driver Operator]), but before you install the driver (create ClusterCSIDriver instance). If you perform this procedure after installing the driver, your volumes will fail to mount into pods.
+
+
+.Prerequisites
+
+* AWS account credentials
+
+.Procedure
+
+To configure AWS EFS CSI Driver Operator with STS:
+
+. Get the ccoctl tool from {product-title} image, which you used to install the cluster with STS (see xref:../../authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc#cco-ccoctl-configuring_cco-mode-sts[Configuring the Cloud Credential Operator utility]).
+
+. Copy the EFS CredentialsRequest (https://github.com/openshift/aws-efs-csi-driver-operator/blob/master/assets/credentials.yaml), save it, and then place it in the `credrequests` directory.
++
+.Example
++
+[source, yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: openshift-aws-efs-csi-driver
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - elasticfilesystem:*
+      effect: Allow
+      resource: '*'
+  secretRef:
+    name: aws-efs-cloud-credentials
+    namespace: openshift-cluster-csi-drivers
+  serviceAccountNames:
+  - aws-efs-csi-driver-operator
+  - aws-efs-csi-driver-controller-sa
+----
+
+.  Use ccoctl to generate a new IAM role in AWS, and create a yaml file for it in the local filesystem (`manifests/openshift-cluster-csi-drivers-aws-efs-cloud-credentials-credentials.yaml`).
++
+[source, terminal]
+----
+$ ccoctl aws create-iam-roles --name=<name> --region=<aws_region> --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests --identity-provider-arn=arn:aws:iam::<aws_account_id>:oidc-provider/<name>-oidc.s3.<aws_region>.amazonaws.com
+----
++
+.Example
++
+[source, terminal]
+----
+$ ./ccoctl aws create-iam-roles --name wduan-0321a-aw-9163 --credentials-requests-dir /home/wduan/01Work/testFile/aws-efs/sts --identity-provider-arn arn:aws:iam::301721915996:oidc-provider/wduan-0321a-aw-9163-oidc.s3.us-east-2.amazonaws.com
```

Comment:
Since in the previous step the reader saved the file to `credrequests/credentials.yaml`, the example should be `-credentials-requests-dir credrequests`.

and to anonymize our project ID and Wei: `--identity-provider-arn arn:aws:iam::123456789012:oidc-provider/aws-efs-oidc.s3.us-east-2.amazonaws.com`